marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -131,19 +126,38 @@ handleQuery (ReceiveWebSocketMessage msg next) = do
           for_ mPlayState \playState -> do
             let
               walletCompanionAppId = view (_walletDetails <<< _companionAppId) playState
-            if plutusAppId == walletCompanionAppId then do
-              -- this is the WalletCompanion app
-              case runExcept $ decodeJSON $ unwrap rawJson of
-                Left decodingError -> addToast $ decodingErrorToast "Failed to parse contract update." decodingError
-                Right companionState -> updateRunningContracts companionState
+
+              marloweAppId = view (_walletDetails <<< _marloweAppId) playState
+            -- if this is the wallet's WalletCompanionApp...
+            if (plutusAppId == walletCompanionAppId) then case runExcept $ decodeJSON $ unwrap rawJson of
+              Left decodingError -> addToast $ decodingErrorToast "Failed to parse contract update." decodingError
+              Right companionState -> handleAction $ PlayAction $ Play.UpdateRunningContracts companionState
             else
-              -- this should be one of the wallet's FollowerContracts (if it isn't, we don't care about it anyway)
-              for_ (lookup plutusAppId $ view _allContracts playState) \contractState -> do
-                case runExcept $ decodeJSON $ unwrap rawJson of
-                  Left decodingError -> addToast $ decodingErrorToast "Failed to parse contract update." decodingError
-                  Right history -> do
+              -- if this is the wallet's MarloweApp...
+              if (plutusAppId == marloweAppId) then
+                -- TODO: in the future the Marlowe app's state will change when there is an error, and we can
+                -- use this to show feedback to the user
+                pure unit
+              -- otherwise this should be one of the wallet's WalletFollowerApps
+              else case runExcept $ decodeJSON $ unwrap rawJson of
+                Left decodingError -> addToast $ decodingErrorToast "Failed to parse contract update." decodingError
+                Right contractHistory -> case contractHistory of
```

Comment:
you can simplify this with:

```purescript
  Left decodingError -> addToast ...
  Right None -> pure unit
  Right (History marloweParams marloweData transactionInputs) -> do
```

Reply:
Good point, I'll change it in the next PR.